devito/types/dimension.py

```diff
@@ -607,6 +607,65 @@ def middle(cls, name, parent, thickness_left, thickness_right, local=False):
                    thickness=((lst, thickness_left), (rst, thickness_right)),
                    local=local)
 
+    @classmethod
+    def intersection(cls, sdims):
```

Comment:
I think this should not be a class method, similar to `Set` that defines intersection as `set().intersectio` not `Set.intersection`.

Reply:
I think it wants to be a class method since you could conceivably want the intersection of several `SubDimension`s, so it would want to be `intersect(subdimensions)` rather than `intersect(self, other)` imo

Reply:
You can still have multiple ones, again similar to a set `set(a).intersection(a,b,c,d,e,f)` there is nothing that prevents it. to have more than one without being a class method